core/src/main/scala/org/http4s/multipart/MultipartParser.scala

```diff
@@ -22,9 +22,7 @@ object MultipartParser {
 
   final case class Out[+A](a: A, tail: Option[ByteVector] = None)
 
-  def parse[F[_]: Sync](
-      boundary: Boundary,
-      headerLimit: Long = 40 * 1024): Pipe[F, Byte, Either[Headers, ByteVector]] = s => {
+  def parse[F[_]: Sync](boundary: Boundary): Pipe[F, Byte, Either[Headers, ByteVector]] = s => {
```

Comment:
I think it's ominous we lost the header limit.

Reply:
Lost because of buffering.

Reply:
Should easily be back with a better implementation.